app/models/ingredient.rb

```diff
@@ -1,3 +1,13 @@
 class Ingredient < ApplicationRecord
-   validates_presence_of :name, :cost
+  validates_presence_of :name, :cost
+  has_many :recipe_ingredients
+  has_many :recipes, through: :recipe_ingredients
+
+  def self.sort_alphabetically(sort)
```

Comment:
I see what you're going for here, and it makes sense and is well-implemented. However know for the other ICs that you don't have to implement this sort of pattern, if it asks you to do a specific ordering you can remove the other kind of ordering on the same object. But this looks good!